src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -73,18 +82,20 @@ export class SharedCredentialsProvider implements CredentialsProvider {
     public validate(): string | undefined {
         const expectedProperties: string[] = []
 
-        if (this.hasProfileProperty(SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN)) {
+        if (hasProfileProperty(this.profile, SHARED_CREDENTIAL_PROPERTIES.ROLE_ARN)) {
```

Comment:
Why was the `hasProfileProperty` method removed from the `Profile` class? It always requires a `Profile` object.